pallets/order-book/src/lib.rs

```diff
@@ -0,0 +1,639 @@
+			Balance = Self::ForeignCurrencyBalance,
+			CustomMetadata = CustomMetadata,
+		>;
+
+		/// CurrencyId of Assets that an order can be made for
+		type AssetCurrencyId: AssetId
+			+ Parameter
+			+ Debug
+			+ Default
+			+ Member
+			+ Copy
+			+ MaybeSerializeDeserialize
+			+ Ord
+			+ TypeInfo
+			+ MaxEncodedLen;
+
+		/// Currency for Reserve/Unreserve with allowlist adding/removal,
+		/// given that the allowlist will be in storage
+		type ReserveCurrency: Currency<Self::AccountId> + ReservableCurrency<Self::AccountId>;
+
+		/// Fee handler for the reserve/unreserve
+		/// Currently just stores the amounts, will be extended to handle
+		/// reserve/unreserve as well in future
+		/// Uses default balance type as opposed to ForeignCurrencyBalance
+		type Fees: Fees<
+			AccountId = <Self as frame_system::Config>::AccountId,
+			Balance = DepositBalanceOf<Self>,
+		>;
+
+		/// Fee Key used to find amount for allowance reserve/unreserve
+		type OrderFeeKey: Get<<Self::Fees as Fees>::FeeKey>;
+
+		/// Balance type for currencies we can place orders for
+		/// Seperate type from Balance in case different type used for other
+		/// currencies, i.e. when Balance is u64, but foreign currencies using
+		/// u128
+		type ForeignCurrencyBalance: Member
+			+ Parameter
+			+ AtLeast32BitUnsigned
+			+ Default
+			+ Copy
+			+ MaxEncodedLen
+			+ FixedPointOperand
+			+ TypeInfo;
+
+		/// Type used for nonce to ensure all orders have unique ID
+		type Nonce: Parameter
+			+ Member
+			+ AtLeast32BitUnsigned
+			+ Default
+			+ Copy
+			+ EnsureAdd
+			+ MaybeSerializeDeserialize
+			+ MaxEncodedLen;
+
+		/// Type for currency orders can be made for
+		type TradeableAsset: MultiReservableCurrency<
+			Self::AccountId,
+			Balance = <Self as pallet::Config>::ForeignCurrencyBalance,
+			CurrencyId = Self::AssetCurrencyId,
+		>;
+
+		/// Type for pallet weights
+		type Weights: WeightInfo;
+	}
+	//
+	// Storage and storage types
+	//
+	/// Order Storage item.
+	/// Contains fields relevant to order information
+	#[derive(Clone, Copy, Debug, Encode, Decode, Eq, PartialEq, MaxEncodedLen, TypeInfo)]
+	pub struct Order<OrderId, AccountId, AssetId, ForeignCurrencyBalance> {
+		pub order_id: OrderId,
+		pub placing_account: AccountId,
+		pub asset_in_id: AssetId,
+		pub asset_out_id: AssetId,
+		/// How many tokens of asset in to buy
+		pub buy_amount: ForeignCurrencyBalance,
+		/// Original buy amount, used for tracking amount fulfilled
+		pub initial_buy_amount: ForeignCurrencyBalance,
+		/// How much currency being purchased (asset in) costs with asset sold
+		/// (asset out)
+		pub price: ForeignCurrencyBalance,
+		/// Minimum amount of an order that can be fulfilled
+		/// for partial fulfillment
+		pub min_fullfillment_amount: ForeignCurrencyBalance,
+		/// Maximum amount of outgoing currency that can be sold
+		pub max_sell_amount: ForeignCurrencyBalance,
+	}
+
+	#[derive(Clone, Copy, Debug, Encode, Decode, Eq, PartialEq, MaxEncodedLen, TypeInfo)]
```

Comment:
You would probably need to add here the following: `#[scale_info(skip_type_params(T))]`

Reply:
Ah, that struct was superfluous, I had missed removing it before.  Removed now.